### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Underdog-01**
This is a proposal for adding the ability to use < input > tags in the package-info.xml file.
It gives mod authors the ability to use HTML5 input and textarea for installing or uninstalling their modifications where the data can be processed in files accessed via the < code > tags.

The DCO.txt file has been manually signed to authorize the use of this code submission.

Please feel free to email me or leave me a message on my forum with any questions or feedback.
Thank you.

### CODE REVIEW

**Comment by Underdog-01** [comment]
This was tested on SMF 2.1 Alpha in which case it works as it should and without error.
Previous commits were deleted as they were branched to the SMF main repository in error. 
Also some on the fly file edits at the time made it look messy so I decided to clean it up.

Notes: Applying these edits as a modification would be inappropriate as it would involve users having to download this mod for developers to make use of it. Therefore adding this set of edits to one of the next releases would be more ideal. 

Without these edits, any author who would put < input > tags in their xml file would create a php error as your current code does not continue the loop for non existent xml tags which imo should be emended. 

@emanuele45 : Thank you for the instruction. I gave permission in the initial comment, however I will create an alias via my github terminal for future pull requests. If I did that permission incorrectly I will edit & do it properly once I figure out the exact procedure as per those instructions.  

It seems as though I will have to redo the instructions for the syntax as I somehow lost the updated one that was more legible. I will do that tomorrow when I have the time.

**Comment by matthew-kerle** [comment]
The biggest problem I see with this right now is that I've not yet committed HTML5 support to SMF 2.1, meaning that SMF 2.1 isn't running HTML5, yet this branch would allow HTML5 inputs.

**Comment by Underdog-01** [comment]
@Labradoodle-360 : It supports some of the newer HTML5 attributes but HTML5 would not be absolutely necessary to use this feature. Mod authors can be informed they should opt to only use the HTML4.1 standards even though the HTML5 ones are available. Imo perhaps HTML4.1 will ignore the HTML5 attributes that are within the < input > brackets when processed anyhow.

  The ones marked as NEW in the link shown below are HTML5 only and the rest are HTML4.1, as I said mod authors can be stressed to use the HTML4.1 standards. Not all attributes were implemented as they were not suitable for the packages template they are being used for. 
ref. http://www.w3schools.com/tags/tag_input.asp

  There is also the option of changing all the templates to HTML 5. All the major browsers support it & the DOCTYPE declaration is a much shorter syntax.

**Comment by matthew-kerle** [comment]
All I was saying is that HTML5 isn't supported by SMF 2.1 yet...I know all about HTML5.

File: new_pkgs_commands-readme.txt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+The form passes the user entered values via the $context['new_inputs'] array where the key values are the name values entered by the author.
+The $context['new_inputs'] array can be processed in any of their <code> files where they can manipulate the logic to do as they wish.
+
+Allowed input types:
+--------------------
+button, checkbox, color, date, datetime, datetime-local, email, hidden, image, month,
+number, password, radio, range, reset, search, submit, tel, text, textarea, time, url, week
+
+
+Allowed input/textare attributes:
```
Comment (live627) [inline new_pkgs_commands-readme.txt:41]: typo

**Comment by ghost** [comment]
ok besides for the typo that's easy fixed are we putting this in?

**Comment by Oldiesmann** [comment]
The biggest issue here is that this is being submitted to the development branch, not the release-2.1 branch. If we're going to pull this in, it needs to be submitted to the release-2.1 branch instead.

File: DCO.txt [hunk 2]
```diff
@@ -22,4 +22,8 @@ Developer's Certificate of Origin 1.1
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
-          this project or the open source license(s) involved.
\ No newline at end of file
+          this project or the open source license(s) involved.
+
+Signed,
+Underdog @ askusaquestion.net     
+Underdog-01 @ github.com
```
Comment (emanuele45) [inline DCO.txt:29]: These three lines should not be here.

**Comment by ghost** [comment]
could someone else submit if Underdog doesnt reply?  But still credit underdog?

**Comment by Underdog-01** [comment]
My apologies ladies & gents. I was busy fixing up that Battle mod & did not notice your comments until today.
I will submit this to the release branch.
I am glad to see you like the idea.

**Comment by Underdog-01** [comment]
Changes committed to Release2.1 branch... done.
I fixed the typos, changed some if/else to switch/case and changed some comparisons to typesafe. 
I'll test it again later today to make sure the switch/case is working properly. 
It does the same thing as if/else I suppose but Imo looks neater (aesthetically).

**Comment by Arantor** [comment]
Closing as per https://github.com/SimpleMachines/SMF2.1/pull/579

**Pull request closed without merging**